### ISSUE TITLE
When I run the spring source code test class, I get an error: No tests fount with test runner 'JUnit 5'

### ISSUE THREAD

**Issue by chang6666** [microsoft/vscode-java-test#1675]
as title,When I run the spring source code test class, I get an error: No tests fount with test runner 'JUnit 5'
![CleanShot 2024-03-27 at 09 41 53@2x](https://github.com/microsoft/vscode-java-test/assets/90826050/21cc2b8d-56e9-4aea-b35f-f6685d9c7511)
eg:spring-beans/src/test/java/org/springframework/beans/AbstractPropertyAccessorTests.java

**Comment by jdneo** [comment]
Does your java file name the same as your test primary class name? e.g.

```
class AbstractPropertyAccessorTests {
...
}
```

**Comment by chang6666** [comment]
@jdneo yes it is the same,I run the same code in idea without any exception,The source code address is at 
    https://github.com/spring-projects/spring-framework/blob/6.1.x/spring-beans/src/test/java/org/springframework/beans/AbstractPropertyAccessorTests.java

**Comment by chang6666** [comment]
@jdneo  
![CleanShot 2024-03-27 at 17 05 52@2x](https://github.com/microsoft/vscode-java-test/assets/90826050/85a65323-94e6-4ba1-830f-44ee5c3135f0)

**Comment by jdneo** [comment]
Where do those `Run Test | Debug Test` come from? I don't understand. We are using gutter icon now. Do you install other java extension?

**Comment by chang6666** [comment]
@jdneoNow that I have removed other test plug-ins, I still get an error
![CleanShot 2024-03-28 at 11 25 36@2x](https://github.com/microsoft/vscode-java-test/assets/90826050/66334968-f506-4e23-bed7-f1438798c43b)

**Comment by chang6666** [comment]
@jdneo Now that I have removed other test plug-ins, I still get an error
![CleanShot 2024-03-28 at 11 30 35@2x](https://github.com/microsoft/vscode-java-test/assets/90826050/88c72306-f03e-467e-bdaf-d9259c7e6598)

**Comment by chang6666** [comment]
@jdneo  Now that I have removed other test plug-ins, I still get an error 
![CleanShot 2024-03-28 at 11 31 51@2x](https://github.com/microsoft/vscode-java-test/assets/90826050/c79cf652-b7ff-4678-ae51-d000fb77a246)

**Comment by jdneo** [comment]
Thank you. I need to make some investigation. Highly likely that we could not handle abstract test class.

**Comment by jrg94** [comment]
Is the extension able to handle test cases in an abstract class? We recently shifted from Eclipse to VSCode. The following example test file matches your example above:

```java
public abstract class QueueTest {

    /**
     * Invokes the appropriate {@code Queue} constructor for the implementation
     * under test and returns the result.
     *
     * @return the new queue
     * @ensures constructorTest = <>
     */
    protected abstract Queue<String> constructorTest();

    /**
     * Invokes the appropriate {@code Queue} constructor for the reference
     * implementation and returns the result.
     *
     * @return the new queue
     * @ensures constructorRef = <>
     */
    protected abstract Queue<String> constructorRef();

    @Test
    public final void testNoArgumentConstructor() {
        /*
         * Set up variables and call method under test
         */
        Queue<String> q = this.constructorTest();
        Queue<String> qExpected = this.constructorRef();
        /*
         * Assert that values of variables match expectations
         */
        assertEquals(qExpected, q);
    } 
}
```

The tests correctly have run buttons next to the line numbers, but clicking them raises an error (i.e., no tests found with testrunner JUnit 4). The class that inherits these tests can be executed just fine. I've been trying to figure out a workaround because my students often click these run buttons and wonder what's wrong. 

I can open another issue if this is not the appropriate location.
